src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts

```diff
@@ -334,102 +335,92 @@ const submitEncryptModeForm: ControllerHandler<
     }
   }
 
-  const submission = new EncryptSubmission({
+  const submissionContent = {
     form: form._id,
     authType: form.authType,
     myInfoFields: form.getUniqueMyInfoAttrs(),
     encryptedContent: incomingSubmission.encryptedContent,
     verifiedContent: verified,
     attachmentMetadata,
     version: req.body.version,
-  })
-
-  let savedSubmission
-  try {
-    savedSubmission = await submission.save()
-  } catch (err) {
-    logger.error({
-      message: 'Encrypt submission save error',
-      meta: {
-        action: 'onEncryptSubmissionFailure',
-        ...createReqMeta(req),
-      },
-      error: err,
-    })
-    return res.status(StatusCodes.BAD_REQUEST).json({
-      message:
-        'Could not send submission. For assistance, please contact the person who asked you to fill in this form.',
-      submissionId: submission._id,
-    })
   }
 
-  const submissionId = String(savedSubmission._id)
-  logger.info({
-    message: 'Saved submission to MongoDB',
-    meta: {
-      ...logMeta,
-      submissionId,
-      formId,
-    },
-  })
-
-  // Fire webhooks if available
-  // To avoid being coupled to latency of receiving system,
-  // do not await on webhook
-  const webhookUrl = form.webhook?.url
-  if (webhookUrl) {
-    void WebhookFactory.sendInitialWebhook(
-      submission,
-      webhookUrl,
-      !!form.webhook?.isRetryEnabled,
-    )
-  }
-
-  // Client secret for stripe payments if payments are enabled
-  let paymentClientSecret
+  // Handle submissions for payments forms
   if (
     form.payments_field?.enabled &&
     form.payments_channel?.channel === PaymentChannel.Stripe
   ) {
-    if (!form.payments_field.amount_cents) {
+    const amount = form.payments_field.amount_cents
+    // Step 1: Create payment without payment intent id and pending submission id.
+    if (!amount || amount <= 0) {
       logger.error({
         message:
-          'Error when creating payment intent, amount is not a positive integer',
-        meta: {
-          submissionId,
-          ...logMeta,
-        },
+          'Error when creating payment: amount is not a positive integer',
+        meta: logMeta,
       })
       return res.status(StatusCodes.INTERNAL_SERVER_ERROR).json({
         message:
           "The form's payment settings are invalid. Please contact the admin of the form to rectify the issue.",
       })
     }
 
+    const payment = new Payment({
+      amount,
+      email: req.body.paymentReceiptEmail,
+    })
+    const paymentId = payment.id
+
+    // Step 2: Create and save pending submission.
+    const pendingSubmission = new EncryptPendingSubmission({
+      ...submissionContent,
+      paymentId,
+    })
+
+    try {
```

Comment:
Using `neverthrow` might be better than all these try/catch statements?

Reply:
Maybe this can be done in future PRs too, but it will be tech debt :') Just a comment since we're not just in "hackathon" mode anymore, right?

Reply:
Agreed. The rest of the controller uses the try/catch style though, and I didn't want to introduce neverthrow here out of the blue. The entire controller is too long and needs refactoring anyway so I figured we can refactor the entire thing once we merge into develop in beta. It's also a comment I added in #6000, feel free to tag on to that ticket!

Reply:
man sounds messy already :') but yup okay!